### ISSUE TITLE
Terrain delete and rename bug

### ISSUE THREAD

**Issue by codymccoy** [HydrologicEngineeringCenter/HEC-FDA#507]
It keeps failing to delete or rename my LCC tif terrain file. The leading hypothesis right now is that it has to do with file size or permissions that might be on the file. I have no problem if i select a different, much smaller, tif.

**Comment by codymccoy** [comment]
@rnugent3 @Brennan1994 Have either of you guys run into this issue. If I import the lower cache creek terrain tif which is a couple gigs and then try to rename or remove the terrain element, it tries for a while and then fails. When it happens again i will try to grab a screen shot. I don't remember exactly what happens. I think it would be helpful to find some really large tifs and see what happens.

**Comment by rnugent3** [comment]
I wonder if it would be helpful to carry out that test for all similar elements? Terrain, hydraulics, structure inventories. I imagine index points and impact areas will always be small.

**Comment by codymccoy** [comment]
I agree. If the issue has to do with file size then it will probably occur for any of them.

**Comment by codymccoy** [comment]
My LCC terrain:
![image](https://user-images.githubusercontent.com/47757848/191852361-fea6f264-f34d-449c-abfb-483e734af2e0.png)

I renamed it from "my ter" to "my ter2" and it did it instantly. I then opened the file location and checked its properties (pic above). I then tried to rename again to "my ter3" and it froze up for a long time and then said this:
![image](https://user-images.githubusercontent.com/47757848/191852525-f66a9ff6-679e-409b-a6f1-ef1b1d30922d.png)

It seems that just have the folder open is locking the folder?

**Comment by codymccoy** [comment]
And in the above situation, it renamed the element in the tree "my ter3" but didn't update the folder. This is a bad situation, if it can't rename the study folder, then we don't want it renaming the child element.

**Comment by codymccoy** [comment]
And now i can't manually rename the folder. Maybe when the rename failed it thinks it is locked from fda?
![image](https://user-images.githubusercontent.com/47757848/191855114-291cde84-a2bd-48b2-8814-ab38be423fdc.png)

**Comment by codymccoy** [comment]
My leading hypothesis of the problem being that file explorer can't be open when renaming seems to be wrong. I did the same thing with index points and file explorer just updates on the fly if you rename or remove.

**Comment by codymccoy** [comment]
I tested with a tif that was 200 MB and it had no problems even with file explorer open. It must have to either be the size of the file (over 2 gigs) or maybe something is off about that file that i am using.

**Comment by codymccoy** [comment]
This seems to work: import the terrain file. Close fda. Open fda and then rename. 
This tells me that maybe there is an issue with importing the terrain. Maybe the file is getting locked during import?

**Comment by codymccoy** [comment]
I updated the renaming logic to only update the element if the folder was renamed successfully:
![image](https://user-images.githubusercontent.com/47757848/192000624-6fa73c0f-658d-4c75-9915-1f4902a2a7ac.png)

This is the logic that does the initial import of the file:
![image](https://user-images.githubusercontent.com/47757848/192004472-feaba11a-bbcd-44c1-95e9-1198d277111c.png)

**Comment by rnugent3** [comment]
I am good with this solution for now, but this seems like a band-aid, right? we'll still have the problem that renaming and deleting folders doesn't always work. We'll let this bug stay in the backlog.
